Brokerages/WebSocketWrapper.cs

```diff
@@ -63,17 +55,110 @@ public void Initialize(string url)
         /// <param name="data"></param>
         public void Send(string data)
         {
-            _wrapped.Send(data);
+            try
+            {
+                if (!IsOpen)
+                {
+                    Log.Trace($"WebSocketWrapper.Send(): Connection not open (IsOpen:{IsOpen}, State:{_wrapped.State}): attempting connection", true);
+
+                    Connect();
+                }
+
+                var sendBuffer = new ArraySegment<byte>(Encoding.UTF8.GetBytes(data));
+
+                _wrapped.SendAsync(sendBuffer, WebSocketMessageType.Text, true, _cancellationTokenSource.Token).SynchronouslyAwaitTask();
```

Comment:
I think we might need some locks around `Send()`, documentation at https://docs.microsoft.com/en-us/dotnet/api/system.net.websockets.clientwebsocket.sendasync?view=netframework-4.8 states `Exactly one send and one receive is supported on each ClientWebSocket object in parallel.` ?